dowhy/gcm/distribution_change_robust.py

```diff
@@ -0,0 +1,915 @@
+                    )
+
+                X_calib, X_train, _, y_train, T_calib, T_train = train_test_split(
+                    X_train, y_train, T_train, train_size=calib_size, stratify=T_train, random_state=split_random_state
+                )
+
+            w_eval, w_train, w_calib = None, None, None
+
+        else:
+            w = np.concatenate((old_data[sample_weight].values.flatten(), new_data[sample_weight].values.flatten()))
+
+            X_train, X_eval, y_train, y_eval, T_train, T_eval, w_train, w_eval = train_test_split(
+                X, y, T, w, train_size=train_size, stratify=T, random_state=split_random_state
+            )
+            X_calib, T_calib, w_calib = None, None, None
+
+            if calibrator is not None:
+                if calib_size == 0.0:
+                    raise ValueError(
+                        "For calibration, calib_size should be either positive and smaller than the number of samples or a float in the (0, 1) range."
+                    )
+
+                X_calib, X_train, _, y_train, T_calib, T_train, w_calib, w_train = train_test_split(
+                    X_train,
+                    y_train,
+                    T_train,
+                    w_train,
+                    train_size=calib_size,
+                    stratify=T_train,
+                    random_state=split_random_state,
+                )
+
+        def set_func(C):
+            return ThetaC(C, warn_th=0.0).est_theta(
+                X_eval,
+                y_eval,
+                T_eval,
+                X_train,
+                y_train,
+                T_train,
+                w_eval=w_eval,
+                w_train=w_train,
+                method=method,  # One of 'regression', 're-weighting', 'MR',
+                regressor=regressor,
+                regressor_args=regressor_args,
+                regressor_kwargs=regressor_kwargs,
+                regressor_fit_kwargs=regressor_fit_kwargs,
+                classifier=classifier,
+                classifier_args=classifier_args,
+                classifier_kwargs=classifier_kwargs,
+                classifier_fit_kwargs=classifier_fit_kwargs,
+                calibrator=calibrator,
+                X_calib=X_calib,
+                T_calib=T_calib,
+                w_calib=w_calib,
+                calibrator_args=calibrator_args,
+                calibrator_kwargs=calibrator_kwargs,
+                calibrator_fit_kwargs=calibrator_fit_kwargs,
+                all_indep=all_indep,
+                crop=crop,
+            )[0]
+
+        attributions = estimate_shapley_values(set_func, len(sorted_var_names), shapley_config)
+
+    if xfit:
+
+        kf = KFold(n_splits=xfit_folds, shuffle=True, random_state=split_random_state)
+        attributions = np.zeros(len(sorted_var_names))
+
+        for train_index, test_index in kf.split(X):
```

Comment:
Why do you need to have the cross validation here? Don't you just need to fit the model on the data? Or is there a specific selection process based on the splitting.

Reply:
There is no cross-validation / model selection, but in the paper we recommend fitting the nuisance functions (regressions and weights) in a train sample, and then using a separate evaluation sample to compute the target parameter, either with simple sample splitting or with cross-fitting.

Reply:
Ah ok. Do you average something here or why do you need the cross-fitting?

Reply:
Yes, we use the train sample to fit the regression / weights, and then we "predict" the multiply-robust score on the evaluation sample and average that (as explained at the bottom of page 11 [here](https://arxiv.org/pdf/2404.08839))

Reply:
Ok got it, thanks for the clarification!